build/Shared/SharedExtensions.cs

```diff
@@ -62,5 +63,68 @@ public static void ForEach<T>(this IEnumerable<T> enumeration, Action<T> action)
             }
         }
 
+
+        /// <summary>
+        /// Helper function to append an <see cref="int"/> to a <see cref="StringBuilder"/>. Calling
+        /// <see cref="StringBuilder.Append(int)"/> directly causes an allocation by first converting the
+        /// <see cref="int"/> to a string and then appending that result:
+        /// <code>
+        /// public StringBuilder Append(int value)
+        /// {
+        ///     return Append(value.ToString(CultureInfo.CurrentCulture));
+        /// }
+        /// </code>
+        ///
+        /// Note that this uses the current culture to do the conversion while <see cref="AppendInt(StringBuilder, int)"/> does
+        /// not do any cultural sensitive conversion.
+        /// </summary>
+        /// <param name="sb">The <see cref="StringBuilder"/> to append to.</param>
+        /// <param name="value">The <see cref="int"/> to append.</param>
+        public static void AppendInt(this StringBuilder sb, int value)
+        {
+            if (value == 0)
+            {
+                sb.Append('0');
+                return;
+            }
+
+            // special case min value since it'll overflow if we negate it
+            if (value == int.MinValue)
+            {
+                sb.Append("-2147483648");
+                return;
+            }
+
+            // do all math with positive integers
+            if (value < 0)
+            {
+                sb.Append('-');
+                value = -value;
+            }
+
+            // upper range of int is 1 billion, so we start dividing by that to get the digit at that position
+            int divisor = 1_000_000_000;
+
+            // remember when we found our first digit so we can keep adding intermediate zeroes
+            bool digitFound = false;
+            while (divisor > 0)
+            {
+                if (digitFound || value >= divisor)
+                {
+                    digitFound = true;
+                    int digit = value / divisor;
+                    value -= digit * divisor;
```

Comment:
Consider using `Math.DivRem` instead. Some (all?) CPU architectures have the divide instruction calculate both the quotient and the remainder, but when you only use one of the two, the other is discarded. Typically, (native) compilers will detect when you do both `a / b` followed by `a % b`, and then only execute a single divide CPU instruction, but the .NET BCL has this `Math.DivRem` method which I assume is also very likely to use the optimized implementation.